crates/extensions/c8y_mapper_ext/src/tests.rs

```diff
@@ -1069,6 +1069,51 @@ async fn mapper_publishes_supported_operations_for_child_device() {
     .await;
 }
 
+#[tokio::test]
+async fn mapping_child_device_dirs_with_forbidden_characters() {
+    // The test assures tedge-mapper checks if there is a directory for operations for child devices, then it reads and
+    // correctly publishes supported operations message for that child on to `c8y/s/us/child1`
+    // and verifies that message.
+    let cfg_dir = TempTedgeDir::new();
+    create_thin_edge_child_operations(&cfg_dir, "my#complex+child2", vec!["c8y_ChildTestOp1"]);
+
+    let (mqtt, _http, _fs, _timer) = spawn_c8y_mapper_actor(&cfg_dir, false).await;
+    let mut mqtt = mqtt.with_timeout(TEST_TIMEOUT_MS);
+    mqtt.skip(6).await; //Skip all init messages
+
+    mqtt.send(MqttMessage::new(
+        &C8yTopic::downstream_topic(),
+        "106,child-one",
+    ))
+    .await
+    .expect("Send failed");
```

Comment:
question: `106` is smartrest for "get children of device", so as I understand we try to get children of `child-one`, but where is this device defined? We don't refer to `child-one` anywhere else in the test, so I'm a bit confused as to why it's there.

Reply:
`105` is the SmartREST message to `get children of device` and the cloud will respond with a `106` message with that list. So here, the idea was to just send a `106` message with a list of child devices that doesn't contain the locally created child device. I could have even sent empty list, to avoid this confusion.

Reply:
Ah yes, `106` is a "subscribe template" according to C8y Smartrest documentation, so it makes sense, a comment like "simulate a response to 105 from cumulocity" will suffice.